File: guides/micronaut-mongodb-asynchronous/java/src/test/java/example/micronaut/MockFruitRepository.java
package example.micronaut;

import io.micronaut.context.annotation.Replaces;
import io.micronaut.context.annotation.Requires;
import io.micronaut.core.async.publisher.Publishers;
import jakarta.inject.Singleton;
import org.reactivestreams.Publisher;
import reactor.core.publisher.Mono;

@Requires(property = "spec.name", value = "FruitControllerValidationTest")
@Singleton
@Replaces(FruitRepository.class)
class MockFruitRepository implements FruitRepository {

    @Override
    public Publisher<Fruit> list() {
        return Publishers.empty();
    }

    @Override
    public Mono<Boolean> save(Fruit fruit) {
        return Mono.just(false);
    }
}

File: guides/micronaut-mongodb-asynchronous/kotlin/src/test/kotlin/example/micronaut/MockFruitRepository.kt
/*
 * Copyright 2017-2024 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package example.micronaut

import io.micronaut.context.annotation.Replaces
import io.micronaut.context.annotation.Requires
import io.micronaut.core.async.publisher.Publishers
import jakarta.inject.Singleton
import org.reactivestreams.Publisher
import reactor.core.publisher.Mono

@Requires(property = "spec.name", value = "FruitControllerValidationTest")
@Singleton
@Replaces(FruitRepository::class)
class MockFruitRepository : FruitRepository {

    override fun list(): Publisher<Fruit> = Publishers.empty()

    override fun save(fruit: Fruit): Mono<Boolean> = Mono.just(false)
}
